src/main/java/com/fouristhenumber/utilitiesinexcess/common/blocks/BlockDrum.java

```diff
@@ -0,0 +1,227 @@
+        this.setHarvestLevel("pickaxe", 1);
+    }
+
+    @Override
+    public TileEntity createNewTileEntity(World world, int meta) {
+        return new TileEntityDrum(capacity);
+    }
+
+    @Override
+    public boolean hasTileEntity(int metadata) {
+        return true;
+    }
+
+    @Override
+    public boolean onBlockActivated(World world, int x, int y, int z, EntityPlayer player, int side, float hitX,
+        float hitY, float hitZ) {
+        if (world.isRemote) {
+            return true;
+        }
+        TileEntity tile = world.getTileEntity(x, y, z);
+        if (tile instanceof TileEntityDrum drum) {
+            if (drum.tank != null && drum.tank.getFluid() != null) {
+                player.addChatMessage(
+                    new ChatComponentText(
+                        "In drum: " + drum.tank.getFluid()
+                            .getLocalizedName() + " " + drum.tank.getFluid().amount + "L"));
+            } else {
+                player.addChatMessage(new ChatComponentText("Drum is empty"));
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public void onBlockPlacedBy(World world, int x, int y, int z, EntityLivingBase placer, ItemStack stack) {
+        super.onBlockPlacedBy(world, x, y, z, placer, stack);
+        TileEntity tile = world.getTileEntity(x, y, z);
+        if (tile instanceof TileEntityDrum drum) {
+            FluidStack fluid = ItemBlockDrum.getFluidFromStack(stack);
+            drum.setFluid(fluid);
+        }
+    }
+
+    @Override
+    public ArrayList<ItemStack> getDrops(World world, int x, int y, int z, int metadata, int fortune) {
+        // We spawn the correct stack on breakBlock(), make it not drop normally.
+        return new ArrayList<>();
+    }
+
+    @Override
+    public void breakBlock(World world, int x, int y, int z, Block block, int meta) {
+
+        ItemStack drop = new ItemStack(this, 1, meta);
+
+        TileEntity tile = world.getTileEntity(x, y, z);
+        if (tile instanceof TileEntityDrum drum) {
+            if (drum.tank.getFluid() != null) {
+                ItemBlockDrum.setFluid(
+                    drop,
+                    drum.tank.getFluid()
+                        .copy());
+            } else {
+                ItemBlockDrum.clearFluid(drop);
+            }
+        }
+
+        float f = 0.7F;
+        double dx = (double) (world.rand.nextFloat() * f) + (double) (1.0F - f) * 0.5D;
+        double dy = (double) (world.rand.nextFloat() * f) + (double) (1.0F - f) * 0.5D;
+        double dz = (double) (world.rand.nextFloat() * f) + (double) (1.0F - f) * 0.5D;
+
+        EntityItem entityItem = new EntityItem(world, x + dx, y + dy, z + dz, drop);
+        world.spawnEntityInWorld(entityItem);
+        world.removeTileEntity(x, y, z);
+        super.breakBlock(world, x, y, z, block, meta);
+    }
+
+    public static class ItemBlockDrum extends ItemBlock implements IFluidContainerItem {
+
+        public static int capacity = -1;
```

Comment:
```suggestion
        public final int capacity;
```
There could be multiple ItemBlockDrum instances if we register a couple of drum types